packit_service/worker/jobs.py

```diff
@@ -207,6 +208,14 @@ def push_initial_metrics(event: Event, handler: JobHandler, build_targets_len: i
         for _ in range(build_targets_len):
             pushgateway.copr_builds_queued.inc()
 
+    # record # of tasks in the celery queue
+    i = celery_app.control.inspect()
+    # >>> i.scheduled()
+    # {'celery@packit-worker-0': [{'eta': '2022-02-11T16:22:53.954247+00:00',...
+    pushgateway.celery_tasks_count = len(
+        [item for workers_tasks in i.scheduled().values() for item in workers_tasks]
+    )
+
```

Comment:
great that we can get this value in this very simple way! Regarding placing the code in `push_initial_metrics`, this function is called only when we create `Task accepted` status, was this intentional or do we want to record these at the beginning of each task?

when comparing the approaches, with the exporter approach I like that the metrics would be independent on running the tasks (e.g. here if I get it correctly, if in some corner case all the workers would be stuck with some tasks for a longer time, we wouldn't have the information about possibly increasing queue size, right?) and also possibility to have richer metrics, on the other hand, if we want to measure only the queue size, this is a simple way to begin with...WDYT?

also, could you please paste a link to the corresponding documentation?

Reply:
you're making a good point: how tightly we want to couple this metric with our actual codebase - and especially, if things go bad, we may not find that out

(re documentation, I just copy-pasted some StackOverflow code O:-)

I checked those exporters you mentioned during your session and they are hooked to celery's events so the metrics should be updated realtime instead of here when the payload is being processed

I guess the solution I propose here is just too hacky and we shouldn't do that